src/docs.js

```diff
@@ -164,9 +165,11 @@ async function downloader(options) {
     credentials: options.credentials,
   });
 
-  let creds = await auth.awsS3Credentials('read-only', options.bucket, options.project + '/');
+  if (!options._s3) {
+    let creds = await auth.awsS3Credentials('read-only', options.bucket, options.project + '/');
+  }
 
-  let s3 = new aws.S3(creds.credentials);
+  let s3 = options._s3 || new aws.S3(creds.credentials);
```

Comment:
The `creds` here is not accessible, since it's scoped to the body of the conditoinal a few lines earlier:
```
     ReferenceError: creds is not defined
```

I think this could work better if shifted around a little:
```javascript
let s3;
if (options._s3) {
  // for testing
  s3 = options._s3;
} else {
  let creds = await auth.awsS3Credentials('read-only', options.bucket, options.project + '/'); 
  s3 = new aws.S3(creds.credentials);
}
```
The bare `let s3` at the beginning scopes that variable to the whole function; it's then set in both branches of the conditional.